### PR TITLE
Send reference to Notify (+ lots of refactoring)

### PR DESCRIPTION
**Pull request by tuzz**
https://trello.com/c/onBqZonX/447-send-reference-to-notify-when-sending-emails

As I was working on this I decided to do some refactoring.

I'd strongly recommend reviewing this commit by commit.

### CODE REVIEW

File: app/providers/notify_provider.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  def initialize(config: EmailAlertAPI.config.notify)
+    api_key = config.fetch(:api_key)
+    base_url = config.fetch(:base_url)
+
+    self.client = Notifications::Client.new(api_key, base_url)
+    self.template_id = config.fetch(:template_id)
+  end
+
+  def call(address:, subject:, body:, reference:)
+    client.send_email(
```
Comment (thomasleese) [inline app/providers/notify_provider.rb]: Do we need to pass back the return value of this?

File: app/services/metric_service.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
+class MetricService
```
Comment (thomasleese) [inline app/services/metric_service.rb]: I would call this `MetricsService`, but don't think it matters much.

File: app/services/metric_service.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  class << self
```
Comment (thomasleese) [inline app/services/metric_service.rb]: Might be good to get https://github.com/alphagov/email-alert-api/blob/master/app/services/email_sender_service.rb#L12 in here too.

File: app/services/metric_service.rb (hunk 2 above)
Comment (tuzz) [reply app/services/metric_service.rb]: renamed

File: app/services/metric_service.rb (hunk 3 above)
Comment (tuzz) [reply app/services/metric_service.rb]: added

File: app/providers/notify_provider.rb (hunk 1 above)
Comment (tuzz) [reply app/providers/notify_provider.rb]: We don't currently use the return value of this method call.

Do you think we should return something?

Comment (thomasleese) [reply app/providers/notify_provider.rb]: Oh, I guess now that we're using `reference` instead of relying on Notify IDs it doesn't matter.

**Pull request merged**